AEPCampaignClassic/Sources/CampaignClassic+PublicAPI.swift

```diff
@@ -16,4 +16,89 @@ import Foundation
+    ///    - token : A unique device token received after registering your app with APNs servers
+    ///    - userKey : A `string` containing the user identifier
+    ///    - additionalParameters : a dictionary of custom key-value pairs to be sent along with the registration call
+    ///    - callback : a block which will be called after the device registration is complete. The callback returns
+    ///                YES if the registration request completed successfully, NO otherwise
+    static func registerDevice(token: Data, userKey: String, additionalParameter: [String: String]?, callback: @escaping (Bool) -> Void) {
+        let deviceInfo = [CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_DEVICE_NAME: UIDevice.current.name,
+                          CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_DEVICE_MODEL: UIDevice.current.model,
+                          CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_OS_NAME: UIDevice.current.systemName]
+
+        var eventData = [CampaignClassicConstants.EventDataKeys.CampaignClassic.REGISTER_DEVICE: true,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_TOKEN: token.hexDescription,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.USER_KEY: userKey,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO: deviceInfo] as [String: Any]
+
+        // attach additional parameters only if they are available
+        if let additionalParameter = additionalParameter {
+            eventData[CampaignClassicConstants.EventDataKeys.CampaignClassic.ADDITIONAL_PARAMETERS] = additionalParameter
+        }
+
+        let event = Event(name: "CampaignClassic Register Device",
+                          type: CampaignClassicConstants.SDKEventType.CAMPAIGN_CLASSIC,
+                          source: EventSource.requestContent,
+                          data: eventData)
+
+        MobileCore.dispatch(event: event, responseCallback: { responseEvent in
+            guard let responseEvent = responseEvent else {
+                callback(false)
+                return
+            }
+
+            guard let registrationResult = responseEvent.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.REGISTRATION_STATUS] as? Bool else {
+                callback(false)
+                return
+            }
+
+            callback(registrationResult)
+
+        })
+    }
+
+    /// Sends tracking information to the configured Adobe Campaign Classic server.
+    /// Use this API to send tracking info on receiving a notification (silent push).
+    ///
+    /// - Parameter trackingInfo :  a dictionary containing `_dId` and `_mId` received in the push message payload, or in the
+    ///                             launch options before opening the application
+    static func trackNotificationClick(trackingInfo: [String: String]) {
```

Comment:
I think we should update the input type to [String: Any] instead of [String: String]. This will allow customers to directly pass `response.notification.request.content.userInfo` to this SDK API when `func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void)` delegate method is invoked upon user interaction.

Reply:
Also suggest renaming `static func trackNotificationClick(withData data: [String: Any])` to better align with [Swift API design guidelines - argument labels](https://www.swift.org/documentation/api-design-guidelines/#argument-labels). What do you think?

Reply:
I like your suggestion to rename.
```
static func trackNotificationReceive(withUserInfo userInfo: [String: Any])
```

what do you think about the term `userInfo`. It becomes clear that they need to pass userInfo from the NoticationContent

Let me know if the current renaming stands good